File: solutions/0148-sort-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        int length = 0;
        ListNode current = head;
        while (current != null) {
            length++;
            current = current.next;
        }
        int[] arr = new int[length];
        current = head;
        int index = 0;
        while (current != null) {
            arr[index++] = current.val;
            current = current.next;
        }
        Arrays.sort(arr);
        ListNode newHead = new ListNode(0);
        ListNode newCurr = newHead;
        for (int i = 0; i < arr.length; i++) {
            newCurr.next = new ListNode(arr[i]);
            newCurr = newCurr.next;
        }
        return newHead.next;
    }
}

File: solutions/0148-sort-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        
        vector<int> sorted;

        ListNode* temp = head;

        while(temp != NULL) {
            sorted.push_back(temp->val);
            temp = temp->next;
        }

        sort(sorted.begin(), sorted.end());

        ListNode* newHead = NULL;
        ListNode* tail = NULL;

        for(int i = 0; i < sorted.size(); i++) {

            ListNode* node = new ListNode(sorted[i]);

            if(newHead == NULL) {
                newHead = node;
                tail = node;
            } else {
                tail->next = node;
                tail = node;
            }
        }

        return newHead;
    }
};
